src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/ProfilerObserver.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Unity.Profiling;
+using UnityEngine;
+
+namespace StateRecorder
+{
+    public struct ProfilerObserverResult
+    {
+        // Amount of memory (in bytes) the operating system reports in use by the application
+        public long? systemUsedMemory;
+
+        // Time spent (in nanoseconds) by the CPU on the main thread on the current frame
+        public long? cpuTime;
```

Comment:
To answer part of the discussion, this for sure needs to be an accumulation since the last tick, not just for the current frame when we captured.

Reply:
I have updated this to to be an accumulation since the last tick, and have renamed the field to `cpuTimeSincePreviousTick` for clarity. The average CPU time per frame can now be calculated by dividing this value by `framesSincePreviousTick`.

Reply:
Side note: The current implementation requires specifying an upper bound on the number of profile frames to keep in memory (MAX_FRAMES_ACCUM). I've set this to 16k based on my perception of what is a reasonable upper bound. As far as I can tell storing this profiling data is not very expensive.

I did initially experiment with periodically flushing the profiling buffer to avoid this requirement, but I was running into issues of double-counting or losing profiling data. So, currently if `framesSinceLastTick` exceeds MAX_FRAMES_ACCUM it will just report `null`.